packages/kit/src/core/sync/create_manifest_data/index.js

```diff
@@ -81,120 +39,160 @@ export default function create_manifest_data({
 		layouts: { [DEFAULT]: default_layout }
 	});
 
+	/** @param {string} id */
+	function tree_node(id) {
+		if (!tree.has(id)) {
+			tree.set(id, {
+				error: undefined,
+				layouts: {}
+			});
+		}
+
+		return /** @type {import('./types').RouteTreeNode} */ (tree.get(id));
+	}
+
 	const routes_base = posixify(path.relative(cwd, config.kit.files.routes));
 	const valid_extensions = [...config.extensions, ...config.kit.moduleExtensions];
 
 	if (fs.existsSync(config.kit.files.routes)) {
-		list_files(config.kit.files.routes).forEach((file) => {
-			const extension = valid_extensions.find((ext) => file.endsWith(ext));
+		list_files(config.kit.files.routes).forEach((filepath) => {
+			const extension = valid_extensions.find((ext) => filepath.endsWith(ext));
 			if (!extension) return;
 
-			const id = file
-				.slice(0, -extension.length)
-				.replace(/(?:^|\/)index((?:@[a-zA-Z0-9_-]+)?(?:\.[a-z]+)?)?$/, '$1');
-			const project_relative = `${routes_base}/${file}`;
+			const project_relative = `${routes_base}/${filepath}`;
+			const segments = filepath.split('/');
+			const file = /** @type {string} */ (segments.pop());
+
+			if (file[0] !== '+') return; // not a route file
```

Comment:
There's gold in them hills.

Reply:
Not sure if this was discussed in that locked thread, but any way we could make those pluses optional for the people that really hate them? I'm not putting random junk in my route folders, so this would be a heck of a lot nicer:

    /routes
       /routes/foo
         /routes/foo/page.svelte
         /routes/foo/layout.svelte

Reply:
> Not sure if this was discussed in that locked thread, but any way we could make those pluses optional for the people that really hate them? I'm not putting random junk in my route folders, so this would be a heck of a lot nicer:
> 
> ```
> /routes
>    /routes/foo
>      /routes/foo/page.svelte
>      /routes/foo/layout.svelte
> ```

Feels like that would defeat the purpose a bit :p

Reply:
Was discussed I see now, deleting comment

Reply:
I am struggling to see why any prefix is needed though. Wouldn't `analyze`ing every file in the directory and skipping irrelevant ones be pretty much the same as skipping all non-plus-prefixed ones?

Reply:
This has been discussed ad nauseam on #5748. See in particular the 'Prefixes' section of https://github.com/sveltejs/kit/discussions/5748#discussioncomment-3282732

Reply:
For the folks following along like myself...

The best rationale I've heard from anyone about the need for prefixes is that the maintainers want a future proof way to identify files of special significance. Without a prefix, they risk breaking ppl's codebases in the future if they introduce a new special/reserved file name.

Reply:
That's one reason, yes — being able to add new features without forcing an otherwise unwarranted semver major is an important design consideration (the "integers are free!!!!" crowd disagrees with that statement, because they don't understand human psychology).

But there are also very good DX reasons:

* you see a `+` file, you immediately know that it's a route file (and if you're unfamiliar with that concept, searching for `+page` etc in the docs will immediately clue you in, without you needing to know any jargon)
* your files are lexically sorted in a very predictable way — child routes, route files, then colocated files. For example if you have a bunch of colocated files, `index.svelte` gets buried...

  ```
  src/routes
  ├ Foo.svelte
  ├ constants.js
  ├ index.svelte
  └ utils.js
  ```

  ...whereas `+page.svelte` sits apart:

  ```
  src/routes
  ├ +page.svelte
  ├ Foo.svelte
  ├ constants.js
  └ utils.js
  ```

* you can find all your route files in your editor very quickly — if you're using VSCode, `Cmd-P` then just type `+`, or start typing the parts of a route followed by `+`. If you start typing `pa` or `in` then you'll get a bunch of false positives, forcing you to type a lot more in order to filter the list. On a large codebase, details like that start to matter.